pkg/xds/generator/prometheus_endpoint_generator.go

```diff
@@ -66,11 +63,40 @@ func (g PrometheusEndpointGenerator) Generate(ctx xds_context.Context, proxy *co
 	envoyAdminClusterName := envoy_names.GetEnvoyAdminClusterName()
 	prometheusListenerName := envoy_names.GetPrometheusListenerName()
 
-	listener, err := envoy_listeners.NewListenerBuilder().
-		Configure(envoy_listeners.InboundListener(prometheusListenerName, prometheusEndpointAddress, prometheusEndpoint.Port)).
-		Configure(envoy_listeners.FilterChain(envoy_listeners.NewFilterChainBuilder().
-			Configure(envoy_listeners.PrometheusEndpoint(prometheusListenerName, prometheusEndpoint.Path, envoyAdminClusterName)))).
-		Build()
+	inbound, err := manager_dataplane.PrometheusInbound(proxy.Dataplane, ctx.Mesh.Resource)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not get prometheus inbound interface")
+	}
+	iface, err := proxy.Dataplane.Spec.GetNetworking().ToInboundInterface(inbound)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not convert inbound interface")
+	}
+
+	// We assume that skipMTLS = nil is the same as false, so we don't break deployments between 0.5 and 0.5.1
+	skipMTLS := prometheusEndpoint.SkipMTLS == nil || prometheusEndpoint.SkipMTLS.Value
+	var listener *envoy_api.Listener
+	if skipMTLS || !ctx.Mesh.Resource.MTLSEnabled() {
```

Comment:
Would it make sense to handle skipMTLS in a separate function?

Reply:
Refactor